src/Common/CommandHelpWriterBase.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Microsoft.PowerShell.PlatyPS.Model;
+
+namespace Microsoft.PowerShell.PlatyPS
+{
+    /// <summary>
+    /// The abstract class for writing the CommandHelp object to a file in markdown format.
+    /// It can be used by any command help writer that writes to markdown format.
+    /// It ensures that all writers will emit all the sections in the proper order.
+    /// </summary>
+    internal abstract class CommandHelpWriterBase : IDisposable
+    {
+        internal string _filePath;
+        internal StringBuilder sb;
+        internal Encoding _encoding;
+
+        internal CommandHelpWriterBase()
+        {
+            _encoding = Encoding.UTF8;
+            _filePath = string.Empty;
+            sb = Constants.StringBuilderPool.Get();
+        }
+
+        /// <summary>
+        /// Write the CommandHelp object to the specified path in the the settings during initialization.
+        /// The file is overwritten and not appended as we do not expect more than one CommandHelp per file.
+        /// </summary>
+        /// <param name="help">CommandHelp object to write.</param>
+        /// <param name="noMetadata">Specify if the metadata header should be written.</param>
+        /// <param name="metadata">Additional metadata that is appended to the standard metadata header. If nometata is true, additional metadata is not written.</param>
+        /// <returns>FileInfo object of the created file</returns>
+        internal FileInfo Write(CommandHelp help, bool noMetadata, Hashtable? metadata = null)
```

Comment:
Do we need a `noMetadata` parameter? Can the decision be made on `metadata is null`